onnxruntime/contrib_ops/cuda/bert/attention_transpose.cu

```diff
@@ -57,11 +84,21 @@ bool LaunchTransCtx(cudaStream_t stream,
     const int H = head_size / 2;
     const float2* input2 = reinterpret_cast<const float2*>(input);
     float2* output2 = reinterpret_cast<float2*>(output);
-    const dim3 block(H, num_heads, 1);
-    TransposeCtx<float2><<<grid, block, 0, stream>>>(H, input2, output2);
+    if (H * num_heads <= 1024) {
+      const dim3 block(H, num_heads, 1);
+      TransposeCtx<float2><<<grid, block, 0, stream>>>(H, input2, output2);
+    } else {
+      const dim3 block(1024 / num_heads, num_heads, 1);
+      TransposeCtxLarge<float2><<<grid, block, 0, stream>>>(H, input2, output2);
+    }
   } else {
-    const dim3 block(head_size, num_heads, 1);
-    TransposeCtx<float><<<grid, block, 0, stream>>>(head_size, input, output);
+    if (head_size * num_heads <= 1024) {
+      const dim3 block(head_size, num_heads, 1);
+      TransposeCtx<float><<<grid, block, 0, stream>>>(head_size, input, output);
+    } else {
+      const dim3 block(1024 / num_heads, num_heads, 1);
```

Comment:
For "1024 / num_heads", it has assumption that num_heads <= 1024, we may need add some checks (maybe in CheckInputs) for this.